File: ParkHere/UI/parkhere_desktop/lib/model/parking_spot.dart
import 'package:json_annotation/json_annotation.dart';
import 'parking_wing.dart';
import 'parking_spot_type.dart';

part 'parking_spot.g.dart';

@JsonSerializable()
class ParkingSpot {
  final int id;
  final String spotCode;
  final int parkingWingId;
  final int parkingSpotTypeId;
  final bool isOccupied;
  final bool isActive;

  final ParkingWing? parkingWing;
  final ParkingSpotType? parkingSpotType;

  ParkingSpot({
    this.id = 0,
    this.spotCode = '',
    this.parkingWingId = 0,
    this.parkingSpotTypeId = 0,
    this.isOccupied = false,
    this.isActive = true,
    this.parkingWing,
    this.parkingSpotType,
  });

  factory ParkingSpot.fromJson(Map<String, dynamic> json) => _$ParkingSpotFromJson(json);
  Map<String, dynamic> toJson() => _$ParkingSpotToJson(this);
}

File: ParkHere/ParkHere.Services/Database/ParkingSpot.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ParkHere.Services.Database
{
    public class ParkingSpot
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(20)]
        public string SpotCode { get; set; } = null!;

        public int ParkingWingId { get; set; }
        public int ParkingSpotTypeId { get; set; }

        [ForeignKey(nameof(ParkingWingId))]
        public ParkingWing ParkingWing { get; set; } = null!;

        [ForeignKey(nameof(ParkingSpotTypeId))]
        public ParkingSpotType ParkingSpotType { get; set; } = null!;

        public bool IsOccupied { get; set; } = false;
        public bool IsActive { get; set; } = true;

    }
}